### DISCUSSION TITLE
Genji & Observable Framework

### DISCUSSION THREAD

**Discussion by mbostock** [pearmini/genji#187] [General]
Hi there! Founder of Observable here. 👋

It’s very cool to see what you’ve accomplished adapting concepts from Observable notebooks (and our open-source libraries such as Inputs and Plot) into a static site generator! I also enjoyed seeing your alternative take on a reactive runtime using `Signal` (more [`Observable`-like](https://github.com/tc39/proposal-observable)) instead of generators (and `Generators.observe`) like we do in the Observable runtime. I’ve also been thinking about using Observables in the Observable runtime (and not only because of the name), though it would be hard for us to do this in a backwards-compatible way.

I’m wondering if you’ve had a chance to check out [Observable Framework](https://observablehq.com/framework/), the static site generator that we [launched in February](https://observablehq.com/blog/observable-2-0)? There’s a lot of overlap between this project and Framework, and in particular we’d love your help building Framework if we can convince you to contribute! And if not, I would still be happy to hear any plans you have for future work on Genji, or things you might like us to add to Framework to make it usable for you.

Thanks for reading, and thanks for sharing your open-source project!

**Comment by pearmini** [comment]
Hi @mbostock ! 

Thank you so much for reaching out. It's an honor to hear from the founder of Observable himself about Genji!

Contributing to Observable Framework is something I'm definitely interested in. I love the idea “Data Loader” as it truly integrates data preparation into the workflow seamlessly. I'm also enthusiastic about the way Framework organizes templates, styles, and logic within a Markdown file. This even makes me feel that Framework is not just a static site generator for data apps; it has the potential to be the framework for web apps, just like [Astro](https://docs.astro.build/en/getting-started/) but more concise, especially if we could define a component with a single Markdown file. As for future plans for Genji, I'm considering adapting it to more SSGs, such as [Docusaurus](https://docusaurus.io/), and will rely on user feedback to determine the direction of its iteration.

Furthermore, I would love to discuss how my contributions could align with the development of Framework, and to hear more about the specific areas where you think I could help, or you can just assign me some issues to help me get started. 

Thank you again for the opportunity, and I'm looking forward to contributing to Observable Framework!
